src/Mvc/Mvc.Core/src/FromRouteAttribute.cs

```diff
@@ -10,6 +10,20 @@ namespace Microsoft.AspNetCore.Mvc;
 /// <summary>
 /// Specifies that a parameter or property should be bound using route-data from the current request.
 /// </summary>
+/// <remarks>
+/// Binds a parameter or property to the value of a route parameter with the same name,
+/// or the name specified in the <see cref="Name"/> property.
+/// The route parameter name is matched against parameter segments of the route pattern case-insensitively.
+///
+/// For more information about parameter binding see
+/// <see href="https://learn.microsoft.com/aspnet/core/fundamentals/minimal-apis/parameter-binding?">Parameter binding</see>.
+/// </remarks>
+/// <example>
+/// In this example, the value of the 'id' route parameter is bound to the parameter.
+/// <code>
+/// app.MapGet("/from-route/{id}", ([FromRoute] string id) => id);
```

Comment:
What happens if you pass a name argument that doesn't match the route template, like `[FromRoute(Name = "notId")]`? Should we document this in the docstring?

Reply:
If the name does not match a parameter in the route template, the build will fail, e.g.:
```
/home/vscode/.nuget/packages/microsoft.extensions.apidescription.server/9.0.0-rc.2.24474.3/build/Microsoft.Extensions.ApiDescription.Server.targets(68,5): error : System.AggregateException: One or more errors occurred. ('notId' is not a route parameter.)
```

I'm not sure this is worth mentioning but happy to add it if you feel strongly about it.